sui_types/src/crypto.rs

```diff
@@ -138,6 +136,11 @@ impl PublicKeyBytes {
     pub fn to_vec(&self) -> Vec<u8> {
         self.0.to_vec()
     }
+    /// Make a Narwhal-compatible public key from a Sui pub.
+    pub fn make_narwhal_public_key(&self) -> Ed25519PublicKey {
```

Comment:
If we deserialize such a public key from a configuration file, or a network message (received from somewhere random on the network), and `from_bytes` fails (a common occurrence, whether malicious or a typo), this will crash the node.

Consider an implementation of [`TryFrom` or `TryInto` instead](https://doc.rust-lang.org/stable/rust-by-example/conversion/try_from_try_into.html)